pkg/skaffold/build/local/docker.go

```diff
@@ -67,14 +74,18 @@ func (b *Builder) retrieveExtraEnv() []string {
 	return b.localDocker.ExtraEnv()
 }
 
-func (b *Builder) dockerCLIBuild(ctx context.Context, out io.Writer, workspace string, a *latest.DockerArtifact, tag string) (string, error) {
+func (b *Builder) dockerCLIBuild(ctx context.Context, out io.Writer, workspace string, a *latest.DockerArtifact, dependencies []*latest.ArtifactDependency, tag string) (string, error) {
 	dockerfilePath, err := docker.NormalizeDockerfilePath(workspace, a.DockerfilePath)
 	if err != nil {
 		return "", fmt.Errorf("normalizing dockerfile path: %w", err)
 	}
 
 	args := []string{"build", workspace, "--file", dockerfilePath, "-t", tag}
-	ba, err := docker.EvalBuildArgs(b.mode, workspace, a)
+	deps, err := build.CreateBuildArgsFromArtifacts(dependencies, b.artifactStore)
+	if err != nil {
+		return "", fmt.Errorf("unable to resolve required artifacts: %w", err)
+	}
+	ba, err := docker.EvalBuildArgs(b.mode, workspace, a, deps)
```

Comment:
I think this is best done above in buildDocker

Reply:
will refactor

Reply:
done.